python/lsst/daf/butler/registry/interfaces/_collections.py

```
@@ -0,0 +1,349 @@
+
+        Notes
+        -----
+        Concurrent registrations of the same collection should be safe; nothing
+        should happen if the types are consistent, and integrity errors due to
+        inconsistent types should happen before any database changes are made
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def find(self, name: str) -> CollectionRecord:
+        """Return the collection record associated with the given name.
+
+        Parameters
+        ----------
+        name : `str`
+            Name of the collection.
+
+        Returns
+        -------
+        record : `CollectionRecord`
+            Object representing the collection, including its type and ID.
+            If ``record/type == CollectionType.RUN``, this will be a
+            `RunRecord` instance.
+
+        Raises
+        ------
+        MissingCollectionError
+            Raised if the given collection does not exist.
+
+        Notes
+        -----
+        Collections registered by another client of the same layer since the
+        last call to `initialize` or `refresh` may not be found.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get(self, key: Any) -> Optional[CollectionRecord]:
+        """Return the collection record associated with the given
+        primary/foreign key value.
+
+        Parameters
+        ----------
+        key
+            Internal primary key value for the collection.
+
+        Returns
+        -------
+        record : `CollectionRecord` or `None`
+            Object representing the collection, including its type and name.
+            If ``record/type == CollectionType.RUN``, this will be a
+            `RunRecord` instance.  `None` if no collection with the given
+            ID exists in this layer.
+
+        Notes
+        -----
+        Collections registered by another client of the same layer since the
+        last call to `initialize` or `refresh` may not be found.
```

Should this method throw an exception for missing collection? My impression that `get()` is a "stronger" version of a `find()` method, usually if someone has a foreign key it means that object exists/existed. OTOH `find()` is a general search by name, it may be more reasonable to re return `None` from that method if nothing is found.

---

Will do.  In keeping with Python conventions, I'll also rename it from `get` to `__getitem__`.